src/util/color/color.h

```diff
@@ -0,0 +1,67 @@
+#ifndef PREDEFINEDCOLORSET_H
+#define PREDEFINEDCOLORSET_H
+
+#include <QColor>
+#include <QHash>
+
+#include "predefinedcolorsset.h"
+#include "util/math.h"
+
+namespace Color {
+    static const PredefinedColorsSet predefinedColorSet = PredefinedColorsSet();
+
+    // algorithm by http://www.nbdtech.com/Blog/archive/2008/04/27/Calculating-the-Perceived-Brightness-of-a-Color.aspx
+    // NOTE(Swiftb0y): please suggest if I should you use other methods
+    // (like the W3C algorithm) or if this approach is to to performance hungry
+    // NOTE: the author did not take alpha transparency into account!
+    static inline int brightness(int red, int green, int blue) {
+        return static_cast<int>(sqrtf(
+            red * red * .241 +
+            green * green * .691 +
+            blue * blue * .068)
+        );
+    };
+
+    static inline int brightness(const QColor& color) {
+        return brightness(color.red(), color.green(), color.red());
+    }
+
+    static inline bool isDimmColor(const QColor& color) {
+        return brightness(color) <= 127;
+    }
+
+    // If the colorToChooseBy is darker than the global threshold,
+    // dimmColor will be returned. Otherwise brightColor will be returned.
+    static inline QColor chooseColorByBrightness(QColor colorToChooseBy, QColor dimmColor , QColor brightColor) {
+        return isDimmColor(colorToChooseBy) ? dimmColor : brightColor;
+    }
+
+    // If the baseColor is darker than the global threshold,
+    // returns a lighter color, otherwise returns a darker color.
+    static inline QColor chooseContrastColor(QColor baseColor) {
+        // Will produce a color that is 60% brighter.
+        static const int iLighterFactor = 160;
+        // We consider a hsv color dark if its value is <= 20% of max value
+        static const int iMinimumValue = 20 * 255 / 100;
+
+        // Convert to Hsv to make sure the conversion only happens once.
+        // QColor::darker() and QColor::lighter() internally convert from and to Hsv if the color is not already in Hsv.
+        baseColor = baseColor.toHsv();
+
+        QColor lightColor;
+        QColor darkColor = baseColor.darker().toRgb();
+        // QColor::lighter() multiplies the HSV Value by some factor. When baseColor is dark, Value is near 0,
+        // thus after multiplication it's still near 0 and we get roughly the same color.
+        // We manually set lightColor in this case.
+        if (baseColor.value() <= iMinimumValue) {
+            lightColor = baseColor;
+            int newValue = iMinimumValue * iLighterFactor / 100;
+            lightColor.setHsl(baseColor.hue(), baseColor.saturation(), newValue);
+        } else {
+            lightColor = baseColor.lighter(iLighterFactor);
+        }
+        QColor contrastColor = chooseColorByBrightness(baseColor, lightColor.toRgb(), darkColor.toRgb());
```

Comment:
You can replace this by 
If (baseColor.value() < 127)

Reply:
Actually, the value component of HSV is not a good measure of brightness. The lightness value of HSL is. But we don't have the HSL representation here anyway.

Reply:
Ah Ok, I understand. 

Can you add a comment that explains our decision?

Something like:
The V value represents the brightness relative to the current color (const H and V).  Here we need to pick a contrast colour compared to all colours. 
The brightness function represents the brightness of the colour compared to black and white.

Reply:
Added a comment.